### ISSUE TITLE
GCC fails if <windows.h> is included before <string>

### ISSUE THREAD

**Issue by Ede123** [Alexpux/MINGW-packages#2463]
Is this a new bug introduced between `mingw-w64-x86_64-gcc-6.2.0-2` and `mingw-w64-x86_64-gcc-6.3.0-2` or am I just doing something stupid?

When compiling the following
```
#include<windows.h>
#include<string>

int main(int argc, char *argv[]){
   return 0;
}
```

I get very ugly errors from gcc (see attached).
If I change the inclusion order of `#include<windows.h>` and `#include<string>` the problem is solved.

[error2.txt](https://github.com/Alexpux/MINGW-packages/files/999381/error2.txt)

**Comment by mati865** [comment]
In short there is conflict between mingw-w64 and libstdc++ `__in` and `__out` after https://gitlab.com/mati865/mingw-w64-mirror/commit/b7f44b677f43c3bb5f771ff5a430206de040c48c

As workaround downgrade mingw-w64-crt and mingw-w64-headers.

**Comment by omichel** [comment]
I confirm I am also affected by this bug.

**Comment by loganmc10** [comment]
I am experiencing the same thing

**Comment by mati865** [comment]
Should be fixed with updated GCC package (available in binary repo).

**Comment by Ede123** [comment]
Already testing...

**Comment by Ede123** [comment]
Problem confirmed fixed (or at least worked around) by 9fbabc02488c69d4ac338de3b615057947912874 and the binary release of  gcc version 6.3.0-3.

@Alexpux and @DavidEGrayson: Thanks for the quick reaction time! Are there already plans for a real fix at this time that does not require to patch almost 3000 lines in libstdc++?

**Comment by Optiligence** [comment]
As i understand, this _is_ the real fix, which is wished to be merged upstream.
